backend/src/services/v2/registry.ts

```diff
@@ -0,0 +1,70 @@
+import fetch from 'node-fetch'
+
+import { UserDoc } from '../../models/v2/User.js'
+import { getAccessToken } from '../../routes/v1/registryAuth.js'
+import config from '../../utils/v2/config.js'
+import { getHttpsAgent } from './http.js'
+
+const registry = config.registry.connection.internal
+const httpsAgent = getHttpsAgent({
+  rejectUnauthorized: !config.registry.connection.insecure,
+})
+
+export interface RepoRef {
+  namespace: string
+  model: string
+}
+
+export interface ImageRef {
+  namespace: string
+  model: string
+  version: string
+}
+
+// Currently limited to a maximum 100 image names
+export async function listModelRepos(user: UserDoc, modelId: string) {
+  const token = await getAccessToken({ id: user.dn, _id: user.dn }, [
+    { type: 'registry', class: '', name: 'catalog', actions: ['*'] },
+  ])
+
+  const authorisation = `Bearer ${token}`
+
+  const catalog = await fetch(`${registry}/v2/_catalog?n=100&last=${modelId}`, {
+    headers: {
+      Authorization: authorisation,
+    },
+    agent: httpsAgent,
+  }).then((res) => res.json() as Promise<{ repositories: Array<string> }>)
+
+  return catalog.repositories.filter((repo) => repo.startsWith(`${modelId}/`))
+}
+
+export async function listImageTags(user: UserDoc, imageRef: RepoRef) {
+  const repo = `${imageRef.namespace}/${imageRef.model}`
+  const token = await getAccessToken({ id: user.dn, _id: user.dn }, [
```

Comment:
What is the difference between 'id' and '_id' ?

Reply:
We used to have the `_id` field represent the internal MongoDB ID and the 'id' field represent the public identifier of a user.  This was confusing, so we now only use the 'dn' field to represent a user.

This `id` and `_id` is to make sure our registry continues working during the beta, as our `registryAuth` is still using the v1 authentication system.